diff --git a/nixos/modules/services/monitoring/prometheus/exporters.nix b/nixos/modules/services/monitoring/prometheus/exporters.nix
index 7b02e815cdbb..b9a169e27885 100644
--- a/nixos/modules/services/monitoring/prometheus/exporters.nix
+++ b/nixos/modules/services/monitoring/prometheus/exporters.nix
@@ -73,6 +73,7 @@ let
     "process"
     "pve"
     "py-air-control"
+    "qbittorrent"
     "redis"
     "restic"
     "rspamd"
@@ -437,6 +438,16 @@ in
       echo "DELUGE_PASSWORD=$(cat ${config.services.prometheus.exporters.deluge.delugePasswordFile})" > /etc/deluge-exporter/password
       '';
     };
+  })] ++ [(mkIf (
+    config.services.prometheus.exporters.qbittorrent.enable &&
+    config.services.prometheus.exporters.qbittorrent.qbittorrentPasswordFile != null
+  ) {
+    system.activationScripts = {
+      qbittorrent-exported.text = ''
+      mkdir -p /etc/qbittorrent-exporter
+      echo "QBITTORRENT_PASS=$(cat ${config.services.prometheus.exporters.qbittorrent.qbittorrentPasswordFile})" > /etc/qbittorrent-exporter/password
+      '';
+    };
   })] ++ (mapAttrsToList (name: conf:
     mkExporterConf {
       inherit name;
